mailpoet/assets/js/src/mailpoet-email-editor-integration/email-sidebar-extension.tsx

```diff
@@ -0,0 +1,142 @@
+    const currentPostType = 'mailpoet_email'; // only for mailpoet_email post-type
+
+    const editedPost = select(coreDataStore).getEditedEntityRecord(
+      'postType',
+      currentPostType,
+      postId,
+    );
+
+    // @ts-expect-error Property 'mailpoet_data' does not exist on type 'Updatable<Attachment<any>>'.
+    const mailpoetData = editedPost?.mailpoet_data || {};
+    void dispatch(coreDataStore).editEntityRecord(
+      'postType',
+      currentPostType,
+      postId,
+      {
+        mailpoet_data: {
+          ...mailpoetData,
+          [name]: value,
+        },
+      },
+    );
+  };
+
+  const subjectHelp = createInterpolateElement(
+    __(
+      'Use personalization tags to personalize your email, or learn more about <bestPracticeLink>best practices</bestPracticeLink> and using <emojiLink>emoji in subject lines</emojiLink>.',
+      'mailpoet',
+    ),
+    {
+      bestPracticeLink: (
+        // eslint-disable-next-line jsx-a11y/anchor-has-content, jsx-a11y/control-has-associated-label
+        <a
+          href="https://www.mailpoet.com/blog/17-email-subject-line-best-practices-to-boost-engagement/"
+          target="_blank"
+          rel="noopener noreferrer"
+        />
+      ),
+      emojiLink: (
+        // eslint-disable-next-line jsx-a11y/anchor-has-content, jsx-a11y/control-has-associated-label
+        <a
+          href="https://www.mailpoet.com/blog/tips-using-emojis-in-subject-lines/"
+          target="_blank"
+          rel="noopener noreferrer"
+        />
+      ),
+    },
+  );
+
+  const previewTextLength = mailpoetEmailData?.preheader?.length ?? 0;
+
+  const preheaderHelp = createInterpolateElement(
+    __(
+      '<link>This text</link> will appear in the inbox, underneath the subject line.',
+      'mailpoet',
+    ),
+    {
+      link: (
+        // eslint-disable-next-line jsx-a11y/anchor-has-content, jsx-a11y/control-has-associated-label
+        <a
+          href={new URL(
+            'article/418-preview-text',
+            'https://kb.mailpoet.com/',
+          ).toString()}
+          key="preview-text-kb"
+          target="_blank"
+          rel="noopener noreferrer"
+        />
+      ),
+    },
+  );
+
+  return (
+    <>
+      <RichTextWithButton
+        attributeName="subject"
+        attributeValue={mailpoetEmailData?.subject || ' '}
+        updateProperty={updateEmailMailPoetProperty}
+        label={__('Subject', 'mailpoet')}
+        labelSuffix={
+          <ExternalLink href="https://kb.mailpoet.com/article/435-a-guide-to-personalisation-tags-for-tailored-newsletters#list">
+            {__('Guide', 'mailpoet')}
+          </ExternalLink>
+        }
+        help={subjectHelp}
+        placeholder={__('Eg. The summer sale is here!', 'mailpoet')}
+      />
+
+      <br />
+
+      <RichTextWithButton
+        attributeName="preheader"
+        attributeValue={mailpoetEmailData?.preheader || 'preheader'}
```

Comment:
This line causes a default value `preheader` instead of the empty value. I would say it is a side effect because the user does not see the initial placeholder.

Reply:
Yeah, it's a side effect. We need a value here; otherwise, the RichTextWithButton component won't [render](https://github.com/mailpoet/mailpoet/blob/c98abf0e3645a2e6c2e90230b8471af4899d62a8/packages/js/email-editor/src/components/personalization-tags/rich-text-with-button.tsx#L100-L102). 

Let me know if you have an idea of what we can do here. If not, we can leave it as is.

Reply:
I checked [the commit](https://github.com/mailpoet/mailpoet/commit/891bc34efa1a290f1fba1486e192a37ee670d325) adding the condition for rendering, and I think we can remove it now. Based on the commit message, the condition should not be relevant anymore. What do you think?

Reply:
I agree. Removed in https://github.com/mailpoet/mailpoet/pull/6089/commits/7c160b8b8783be3c2f42b1e69b40ce33535fc3e4